syncing/src/test/kotlin/maru/syncing/beaconchain/CLSyncServiceImplTest.kt

```diff
@@ -0,0 +1,493 @@
+    private val key2 =
+      "0x0802122100f3d2fffa99dc8906823866d96316492ebf7a8478713a89a58b7385af85b088a1"
+        .fromHexToByteArray()
+  }
+
+  private var port1: UInt = 0u
+  private var port2: UInt = 0u
+  private lateinit var signatureAlgorithm: SignatureAlgorithm
+  private lateinit var keypair: KeyPair
+  private lateinit var beaconChain1: BeaconChain
+  private lateinit var beaconChain2: BeaconChain
+  private lateinit var validators: Set<Validator>
+  private lateinit var p2pNetwork1: P2PNetworkImpl
+  private lateinit var p2pNetwork2: P2PNetworkImpl
+  private lateinit var clSyncService: CLSyncServiceImpl
+  private lateinit var peerLookup: PeerLookup
+
+  @BeforeEach
+  fun setUp() {
+    signatureAlgorithm = SignatureAlgorithmFactory.getInstance()
+    keypair = signatureAlgorithm.generateKeyPair()
+    validators = setOf(Validator(Util.publicKeyToAddress(keypair.publicKey).toArray()))
+
+    val genesisTimestamp = DataGenerators.randomTimestamp()
+    val (genesisBeaconState, genesisBeaconBlock) = genesisState(genesisTimestamp, validators)
+    beaconChain1 = spy(InMemoryBeaconChain(genesisBeaconState, genesisBeaconBlock))
+    beaconChain2 = spy(InMemoryBeaconChain(genesisBeaconState, genesisBeaconBlock))
+
+    port1 = findFreePort()
+    port2 = findFreePort()
+    p2pNetwork1 = createNetwork(beaconChain1, key1, port1)
+    p2pNetwork2 = createNetwork(beaconChain2, key2, port2)
+
+    createBlocks(
+      beaconChain = beaconChain2,
+      genesisBeaconBlock = genesisBeaconBlock,
+      genesisTimestamp = genesisBeaconBlock.beaconBlock.beaconBlockHeader.timestamp,
+      validators = validators,
+      signatureAlgorithm = signatureAlgorithm,
+      keypair = keypair,
+    )
+    peerLookup = spy(p2pNetwork1.getPeerLookup())
+    clSyncService =
+      CLSyncServiceImpl(
+        beaconChain = beaconChain1,
+        executorService = Executors.newCachedThreadPool(),
+        validatorProvider = StaticValidatorProvider(validators),
+        allowEmptyBlocks = true,
+        peerLookup = peerLookup,
+        besuMetrics = TestMetricsSystemAdapter,
+        metricsFacade = TestMetricsFacade,
+        pipelineConfig = Config(blocksBatchSize = 10u, blocksParallelism = 1u),
+      )
+
+    try {
+      p2pNetwork1.start()
+      p2pNetwork2.start()
+      p2pNetwork1.addStaticPeer(createPeerAddress(port2))
+
+      awaitUntilAsserted { assertNetworkHasPeers(network = p2pNetwork1, peers = 1) }
+      awaitUntilAsserted { assertNetworkHasPeers(network = p2pNetwork2, peers = 1) }
+    } catch (e: Exception) {
+      p2pNetwork1.stop()
+      p2pNetwork2.stop()
+      throw IllegalStateException("Failed to start P2P networks", e)
+    }
+  }
+
+  @AfterEach
+  fun tearDown() {
+    p2pNetwork1.stop()
+    p2pNetwork2.stop()
+  }
+
+  @Test
+  fun `chain sync downloads and imports blocks from another node`() {
+    val synced = AtomicBoolean(false)
+    clSyncService.setSyncTarget(100uL)
```

Comment:
We should either name the magic number `100uL` and define it as a class field for reuse or you can just take the `beaconChain2.getLatestBeaconState().latestBeaconBlockHeader.number` with a proper name like `expectedChainHead` and use it